update changelog for v0.42.4 (#9083)
cosmos/cosmos-sdk: CHANGELOG.md

# Changelog

## [v0.42.4](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.42.4) - 2021-04-08

### Client Breaking Changes

> * [\#9026](https://github.com/cosmos/cosmos-sdk/pull/9026) The `tx sign` and `tx sign-batch` CLI commands use SIGN_MODE_DIRECT by default for local pubkeys. For multisigs and ledger keys, the default LEGACY_AMINO_JSON is kept.

Reviewer: ```suggestion
* [\#9026](https://github.com/cosmos/cosmos-sdk/pull/9026) By default, the `tx sign` and `tx sign-batch` CLI commands use SIGN_MODE_DIRECT to sign transactions for local pubkeys. For multisigs and ledger keys, the default LEGACY_AMINO_JSON is used.
```

Reviewer: I chatted with @alessio about this, but I feel like there should be a statement that identifies this change as a backport, seems odd to see a non-breaking change in this section

### Bug Fixes

* (gRPC) [\#9015](https://github.com/cosmos/cosmos-sdk/pull/9015) Fix invalid status code when accessing gRPC endpoints.

> * [\#9026](https://github.com/cosmos/cosmos-sdk/pull/9026) Fix bug of `gentx` command not working with ledger keys.

Reviewer: ```suggestion
* [\#9026](https://github.com/cosmos/cosmos-sdk/pull/9026) Fixed the bug that caused the `gentx` command to fail for Ledger keys.
```

### Improvements

> * [\#9081](https://github.com/cosmos/cosmos-sdk/pull/9081) Upgrade tendermint to v0.34.9, which includes a fix of a security issue with Tendermint light clients

Reviewer: ```suggestion
* [\#9081](https://github.com/cosmos/cosmos-sdk/pull/9081) Upgrade Tendermint to v0.34.9 that includes a security issue fix for Tendermint light clients.
```
